plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitManager.java

```diff
@@ -108,6 +112,31 @@ public ConnectorSplitSource getSplits(
         return new ClassLoaderSafeConnectorSplitSource(splitSource, IcebergSplitManager.class.getClassLoader());
     }
 
+    private Scan<?, FileScanTask, CombinedScanTask> getScan(IcebergMetadata icebergMetadata, Table icebergTable, IcebergTableHandle table, ExecutorService executor)
+    {
+        Long fromSnapshot = icebergMetadata.getIncrementalRefreshFromSnapshot(table.getSchemaTableName()).orElse(null);
+        if (fromSnapshot != null) {
```

Comment:
Now that we've restricted the scope to single source table MVs, I think this shouldn't be a problem.
If the MV's snapshot summary does not contain the single source table's `fromSnapshot`, we don't add anything to the `fromSnapshotsForRefresh` map and therefore it will be an empty map. So the `disableIncrementalRefresh()` would be a noop.